### PR TITLE
fix: update MongoDB schema converter to remove invalid fields, add integration test

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-9110

Adds an integration test that creates a validation rule with all types and some edge cases (nested empty arrays/objects).

Opening as a draft, I need to fix the error on the test.

### CODE REVIEW

File: test/all-bson-types-fixture.ts [hunk 1]
```diff
@@ -49,9 +49,29 @@ export const allBSONTypesDoc = {
...
     md5: Binary.createFromBase64('c//SZESzTGmQ6OfR38A11A==', 5), // 5
     encrypted: Binary.createFromBase64('c//SZESzTGmQ6OfR38A11A==', 6), // 6
-    compressedTimeSeries: Binary.createFromBase64('c//SZESzTGmQ6OfR38A11A==', 7), // 7
+    compressedTimeSeries: new Binary(
+      Buffer.from(
+        'CQCKW/8XjAEAAIfx//////////H/////////AQAAAAAAAABfAAAAAAAAAAEAAAAAAAAAAgAAAAAAAAAHAAAAAAAAAA4AAAAAAAAAAA==',
+        'base64'
+      ),
+      7
+    ), // 7
```
Comment (paula-stacho) [inline test/all-bson-types-fixture.ts:58]: just curious what is the difference

File: src/schema-converters/internalToMongoDB.ts [hunk 2]
```diff
@@ -46,15 +46,17 @@ async function parseType(type: SchemaType, signal?: AbortSignal): Promise<MongoD
...
-    case 'Document':
-      Object.assign(schema,
-        await parseFields((type as DocumentSchemaType).fields, signal)
-      );
-      break;
+
+  if (type.bsonType === 'Array') {
+    const items = await parseTypes((type as ArraySchemaType).types);
+    // Don't include empty bson type arrays (it's invalid).
+    if (!items.bsonType || items.bsonType?.length > 0) {
```
Comment (paula-stacho) [inline src/schema-converters/internalToMongoDB.ts:53]:
```suggestion
    if (!!items.bsonType && items.bsonType?.length > 0) {
```
or?

Comment (Anemy) [reply src/schema-converters/internalToMongoDB.ts:53]: This is probably a habit of mine from pre-typescript days. If we remove the or we we'd be comparing `undefined > 0` which is a check I like to avoid as they're differing types (it is still false though). If it's alright with you I'd like to keep it as is, although I could remove the `?` from the 2nd part of it.

Comment (paula-stacho) [reply src/schema-converters/internalToMongoDB.ts:53]: but then shouldn't it be `!!items.bsonType && items.bsonType?.length > 0`? because now if it's falsy we're still going in and adding the items.

Comment (Anemy) [reply src/schema-converters/internalToMongoDB.ts:53]: Ah I think that comment on this line could be better. What's happening here is that we want to omit it when it's an empty array. When `bsonType` doesn't exist on items it's one of the `anyOf` types `items`.

Comment (paula-stacho) [reply src/schema-converters/internalToMongoDB.ts:53]: Ah I see, thanks!

**Review by paula-stacho — APPROVED** [review]
